src/sentry/migrations/0287_backfill_snubaquery_environment.py

```diff
@@ -0,0 +1,70 @@
+# Generated by Django 2.2.27 on 2022-04-06 22:58
+
+from django.db import migrations
+
+from sentry.new_migrations.migrations import CheckedMigration
+from sentry.utils.query import RangeQuerySetWrapperWithProgressBar
+
+
+def backfill_snubaquery_environment(apps, schema_editor):
+    # the current snuba_query.environment_id could be referencing an environment it shouldn't be:
+    # during project transfer, the AlertRule changed organization_id's but some work wasn't done:
+    # 1. Conditionally create the environment from the old org to the new org
+    # 2. Update the environment_id for the SnubaQuery to reference the new org environment
+    # So now we have SnubaQueries that reference(fk to Environment) environments from the old org.
+
+    SnubaQuery = apps.get_model("sentry", "SnubaQuery")
+    Environment = apps.get_model("sentry", "Environment")
+
+    for snuba_query in RangeQuerySetWrapperWithProgressBar(SnubaQuery.objects.all()):
+        if not snuba_query.environment_id:
+            continue
+
+        snuba_env = Environment.objects.filter(snubaquery__id=snuba_query.id).get()
+        envs_by_project = Environment.objects.filter(
+            environmentproject__project__querysubscription__snuba_query=snuba_query
+        )
```

Comment:
It might be safer to do `Environment.objects.filter(organization__project__querysubscription__snuba_query=snuba_query).distinct()`. Just in case the environment is for some reason not specifically linked to a project, but exists in the org.